catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/ValidateCommand.java

```diff
@@ -47,178 +55,104 @@
-                prettyPrint(prefix + INDENT + LAST_ELEMENT, "None");
-            } else {
-                printList(prefix + INDENT, e.getErrors(), ERROR_COLOR);
-            }
+        QueryImpl query = new QueryImpl(filter);
+        ThreadContext.bind(Security.getInstance()
+                .getSystemSubject());
 
-            // We know that warnings come last, so we format each line knowing that it's the last
-            // in its hierarchy.
-            prettyPrint(prefix + LAST_ELEMENT, WARNING_TITLE);
-            if (e.getWarnings() == null) {
-                prettyPrint(prefix + LAST_INDENT + LAST_ELEMENT, "None");
-            } else {
-                printList(prefix + LAST_INDENT, e.getWarnings(), WARNING_COLOR);
+        SourceResponse response = catalog.query(new QueryRequestImpl(query));
+        List<Result> resultList = response.getResults();
+        if (resultList != null && resultList.size() > 0) {
+            for (Result r : resultList) {
+                if (r.getMetacard() != null) {
+                    results.add(r.getMetacard());
+                }
             }
-
-        } catch (RuntimeException e) {
-            prettyPrint(prefix + LAST_ELEMENT,
-                    "Unhandled exception while running validator. Check the log for the stack trace and fix the validator.",
-                    ERROR_COLOR);
-            LOGGER.error("Unhandled exception while trying to validate.", e);
         }
-    }
 
-    private void prettyPrint(String prefix, String message) {
-        prettyPrint(prefix, message, COLOR_DEFAULT);
+        return results;
     }
 
-    private void prettyPrint(String prefix, String message, String color) {
-
-        // We want to get rid of all whitespace and newlines to make everything look uniform.
-        console.println(prefix + color + message.replaceAll("[\r\n\t ]+", " ") + COLOR_DEFAULT);
-    }
+    private Collection<File> getFiles() throws FileNotFoundException {
+        File file = new File(path);
 
-    private void printList(String prefix, List<String> messages, String color) {
+        if (!file.exists()) {
+            throw new FileNotFoundException(String.format("File or directory %s does not exist",
+                    path));
+        }
 
-        // Using an iterator instead of a for loop to make sure the last element
-        // in the list is printed differently for console readability.
-        Iterator<String> iterator = messages.iterator();
-        while (iterator.hasNext()) {
-            String message = iterator.next();
-            if (iterator.hasNext()) {
-                prettyPrint(prefix + ELEMENT, message, color);
-            } else {
-                prettyPrint(prefix + LAST_ELEMENT, message, color);
-            }
+        Collection<File> files;
+        if (file.isFile()) {
+            files = Collections.singletonList(file);
+        } else {
+            files = FileUtils.listFiles(file, null, true);
```

Comment:
I think all of that would be useful. Also specifying if you want to recurse directories or not could be helpful as well.